File: app/src/main/java/com/cristhian/gametobeat/ui/list/GameListPageAdapter.java
package com.cristhian.gametobeat.ui.list;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class GameListPageAdapter extends FragmentStatePagerAdapter {

    private Context mContext;
    private List<GameListFragment> mFragmentList;

    public GameListPageAdapter(FragmentManager fm, Context context) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mFragmentList.get(position).getTitle();
    }

    public void addFragment(GameListFragment fragment) {
        if (mFragmentList == null) {
            mFragmentList = new ArrayList<>();
        }
        mFragmentList.add(fragment);
    }
}

File: app/src/main/java/com/cristhian/gametobeat/ui/list/GameListPageAdapter.kt
package com.cristhian.gametobeat.ui.list

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

class GameListPageAdapter(fragmentManager: FragmentManager) : FragmentStatePagerAdapter(fragmentManager) {

    private val fragmentList: ArrayList<GameListFragment> = arrayListOf()

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentList[position].getTitle()
    }

    fun addFragment(fragment: GameListFragment) {
        fragmentList?.add(fragment)
    }
}